### PR TITLE
INFRA-336-userregistry-k8s-setup-kubernetes

### PR DESCRIPTION
**Pull request by diegolagospagopa**
### List of changes

setup kubernetes cluster

### Type of changes

- [x] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [x] DEV
- [ ] UAT
- [ ] PROD

### CODE REVIEW

File: src/.env/dev/kubernetes.tfvars [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+configmaps_uservice-party-management = {
+  JAVA_OPTS                                         = "-javaagent:/applicationinsights-agent.jar"
+  APPLICATIONINSIGHTS_INSTRUMENTATION_LOGGING_LEVEL = "OFF"
+}
+
+configmaps_uservice-party-process = {
+  JAVA_OPTS                                         = "-javaagent:/applicationinsights-agent.jar"
+  APPLICATIONINSIGHTS_INSTRUMENTATION_LOGGING_LEVEL = "OFF"
+  USER_REGISTRY_MANAGEMENT_URL                      = "https://api.dev.user-registry.pagopa.it/pdnd-interop-uservice-user-registry-management/0.1" // TODO
+}
```
Comment (pasqualedevita) [inline src/.env/dev/kubernetes.tfvars]: i think that is not related to user registry

File: src/.env/uat/kubernetes.tfvars [hunk 2]
```diff
@@ -0,0 +1,59 @@
+prefix = "usrreg"
+
+# ingress
+nginx_helm_version="3.31.0"
+ingress_replica_count    = "2"
+ingress_load_balancer_ip = "20.82.206.96"
```
Comment (pasqualedevita) [inline src/.env/uat/kubernetes.tfvars:6]: why a public ip?

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: src/.env/dev/kubernetes.tfvars [hunk 3]
```diff
@@ -0,0 +1,59 @@
+prefix = "usrreg"
+
+# ingress
+nginx_helm_version="3.31.0"
+ingress_replica_count    = "2"
+ingress_load_balancer_ip = "20.82.206.96"
```
Comment (pasqualedevita) [inline src/.env/dev/kubernetes.tfvars:6]:
```suggestion
ingress_load_balancer_ip = "10.1.0.250"
```

File: src/k8s/02_ingress_controller.tf [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+module "nginx_ingress" {
+
+  source  = "terraform-module/release/helm"
+  version = "2.7.0"
+
+  namespace  = kubernetes_namespace.usrreg.metadata[0].name
+  repository = "https://kubernetes.github.io/ingress-nginx"
+  app = {
+    name          = "ingress-nginx"
+    version       = "3.33.0"
```
Comment (pasqualedevita) [inline src/k8s/02_ingress_controller.tf]: use >= 4.0.6
https://pagopa.atlassian.net/browse/INFRA-254

File: src/k8s/03_serviceaccounts.tf [hunk 5]
```diff
@@ -0,0 +1,79 @@
+# resource "kubernetes_service_account" "azure_devops" {
```
Comment (pasqualedevita) [inline src/k8s/03_serviceaccounts.tf]: enable this

File: src/k8s/99_variables.tf [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+variable "ingress_load_balancer_ip" {
+  type = string
+}
+
+variable "default_service_port" {
+  type    = number
+  default = 8080
+}
+
+variable nginx_helm_version {
```
Comment (pasqualedevita) [inline src/k8s/99_variables.tf]:
```suggestion
variable "nginx_helm_version" {
```

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: src/k8s/02_ingress_controller.tf (hunk 4 above)
Comment (diegolagospagopa) [reply src/k8s/02_ingress_controller.tf]: DONE

File: src/.env/dev/kubernetes.tfvars (hunk 3 above)
Comment (diegolagospagopa) [reply src/.env/dev/kubernetes.tfvars:6]: The loadbalancer must be this public ip for DEV

File: src/.env/dev/kubernetes.tfvars (hunk 1 above)
Comment (diegolagospagopa) [reply src/.env/dev/kubernetes.tfvars]: Removed all references

File: src/.env/uat/kubernetes.tfvars (hunk 2 above)
Comment (diegolagospagopa) [reply src/.env/uat/kubernetes.tfvars:6]: in UAT will be change for private

**Review by pasqualedevita — APPROVED** [review]
